### PR TITLE
 Fix #1601 Allow strings to have trailing comments

### LINKED ISSUE #1601: YAML strings fail with trailing comments
# Summary

YAML string highlighting does not work when a string has a comment on the same line.

# Example

Using the following example:

```html
<html>
    <head>
        <link href="prism.css" rel="stylesheet" />
    </head>
    <body>
        <script src="prism.js"></script>
        <pre>
            <code class="language-yaml">
hello:
    - "world" # test
    - "world"
            </code>
        </pre>
    </body>
</html>
```

I'm testing this using version 1.15.0, from this link: https://prismjs.com/download.html#themes=prism&languages=markup+css+clike+javascript+yaml

### PR DESCRIPTION
**Pull request by Chris-Johnston**
Fixes #1601 

Updates the string regex to allow for strings that may have a trailing '#'. This uses a different pattern from other regex patterns, because only the string regex pattern is greedy, while the others are not.

# Example

Using the example code from #1601 , I get this result:

### CODE REVIEW

File: components/prism-yaml.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ Prism.languages.yaml = {
 		alias: 'important'
 	},
 	'string': {
-		pattern: /([:\-,[{]\s*(?:![^\s]+)?[ \t]*)("|')(?:(?!\2)[^\\\r\n]|\\.)*\2(?=[ \t]*(?:$|,|]|}))/m,
+		pattern: /([:\-,[{]\s*(?:![^\s]+)?[ \t]*)("|')(?:(?!\2)[^\\\r\n]|\\.)*\2(?=[ \t]*(?:$|,|]|}|(?:\s*#)))/m,
```
Comment (RunDevelopment) [inline components/prism-yaml.js]: I meant that you don't need the group at all...
Sorry for not making myself clear enough.

Comment (Chris-Johnston) [reply components/prism-yaml.js]: My mistake, I had assumed that a grouping would be necessary. I just tried it and you are right.

**Comment by RunDevelopment** [comment]
Perfect!

@mAAdhaTTah I think we are ready to merge.

**Comment by mAAdhaTTah** [comment]
Thank you for the contribution!

**Comment by plaindocs** [comment]
Thanks @Chris-Johnston @RunDevelopment @mAAdhaTTah

**Pull request merged**